### PR TITLE
[AUD-1682] Support search deep linking properly

### PR DESCRIPTION
**Pull request by raymondjacobson**
### Description

Support search deep linking properly for profiles & collections

### Dragons

*Is there anything the reviewer should be on the lookout for? Are there any dangerous changes?*

### How Has This Been Tested?

*Please describe the tests that you ran to verify your changes. Provide repro instructions & any configuration.*

Locally w/ simulator against staging

### How will this change be monitored?

*For features that are critical or could fail silently please describe the monitoring/alerting being added.*

Staging rollout

### CODE REVIEW

File: packages/mobile/src/components/artist-card/ArtistCard.tsx [hunk 1]
```diff
@@ -15,18 +15,22 @@ const formatProfileCardSecondaryText = (followers: number) => {
...
 
-export const ArtistCard = ({ artist, style }: ArtistCardProps) => {
+export const ArtistCard = ({ artist, fromPage, style }: ArtistCardProps) => {
   const { handle } = artist
   const navigation = useNavigation()
   const handlePress = useCallback(() => {
     navigation.push({
       native: { screen: 'Profile', params: { handle } },
-      web: { route: handle, fromPage: EXPLORE_PAGE }
+      web: { route: handle, fromPage: fromPage || EXPLORE_PAGE }
```
Comment (dylanjeffers) [inline packages/mobile/src/components/artist-card/ArtistCard.tsx:31]: maybe set default in props?

File: packages/mobile/src/components/collection-card/CollectionCard.tsx [hunk 2]
```diff
@@ -18,19 +17,26 @@ const formatPlaylistCardSecondaryText = (saves: number, tracks: number) => {
...
+  collection,
+  fromPage,
+  style
+}: CollectionCardProps) => {
+  const navigation = useNavigation()
+  const handlePress = useCallback(() => {
+    const collectionRoute = getCollectionRoute(collection)
+    navigation.push({
+      native: { screen: 'Collection', params: { id: collection.playlist_id } },
+      web: { route: collectionRoute, fromPage: fromPage || EXPLORE_PAGE }
```
Comment (dylanjeffers) [inline packages/mobile/src/components/collection-card/CollectionCard.tsx:37]: same i guess?

**Review by dylanjeffers — APPROVED** [review]
nice!

File: packages/mobile/src/components/artist-card/ArtistCard.tsx (hunk 1 above)
Comment (raymondjacobson) [reply packages/mobile/src/components/artist-card/ArtistCard.tsx:31]: honestly will just remove default. makes no sense since i moved the files

**Pull request merged**